### PR TITLE
Sync from LFortran's IntrinsicFunctions

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
LFortran: https://github.com/lfortran/lfortran/pull/1485

cc @czgdp1807

### CODE REVIEW

File: integration_tests/intrinsics_01.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from numpy import array
+
+def any_01() -> None:
+    x: bool[2]
+
+    x= array([False, False])
+    assert not any(x)
+
+    x = array([False, True])
+    assert any(x)
```
Comment (czgdp1807) [inline integration_tests/intrinsics_01.py:10]: I think Python's any accepts not only arrays but anything like lists, tuples. Also, it doesn't accepts arguments like `dim` and `mask`.

Comment (Thirumalai-Shaktivel) [reply integration_tests/intrinsics_01.py:10]: I wanted this: https://github.com/lfortran/lfortran/pull/1485 changes in LPython, so I submitted a PR with these changes.

We will do one thing, we will skip `any` for now and implement it later.

**Review by czgdp1807 — APPROVED** [review]

**Comment by Thirumalai-Shaktivel** [comment]
@czgdp1807 should I keep the tests or skip `any` for now?

**Comment by czgdp1807** [comment]
Feel free to keep it.

**Comment by Thirumalai-Shaktivel** [comment]
Thank you! I will merge this PR and work on `Partition`.

**Pull request merged**